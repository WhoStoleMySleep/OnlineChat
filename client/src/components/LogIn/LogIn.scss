.log-in {
  width: fit-content;
  margin: 20px 20px 0 auto;

  &__open-form {
    border: none;
    background: none;
    background-color: rgb(255, 174, 0);
    padding: 13px 40px;
    display: block;
    font-size: 19px;
    border-radius: 4px;
    font-weight: bold;
    box-sizing: border-box;
    color: rgb(255, 249, 213);
  }

  &__form {
    max-width: 600px;
    height: 200px;
    width: 100%;
    position: absolute;
    left: calc(50% - 300px);
    top: calc(50% - 100px);
    background: rgba(0, 0, 0, 0.603);
    border-radius: 8px;
    padding-top: 40px;
    box-sizing: border-box;
    display: none;
    box-shadow: 0px 6px 5px rgba(0, 0, 0, 0.05);
    outline: 100vw solid rgba(0, 0, 0, 0.8);

    &.open {
      display: block;
    }
  }

  &__author {
    width: 350px;
    height: 38px;
    border: none;
    background: none;
    outline: none;
    background-color: rgba(0, 0, 0, 0.603);
    border-radius: 4px;
    margin-bottom: 20px;
    padding-left: 15px;
    box-sizing: border-box;
    caret-color: #fff;
    color: rgba(255, 255, 255, 0.603);
    transition-duration: 50ms;
    opacity: 0.6;

    &:focus {
      opacity: 1;
    }

    &::placeholder {
      color: rgba(255, 255, 255, 0.322);
    }
  }

  &__close-form {
    border: none;
    background: none;
    background-color: rgba(0, 0, 0, 0.603);
    padding: 13px 40px;
    font-size: 19px;
    border-radius: 4px;
    font-weight: bold;
    box-sizing: border-box;
    color: rgb(255, 249, 213);
    opacity: 0.6;
    transition-duration: 50ms;

    &:hover {
      opacity: 1;
    }
  }
}
