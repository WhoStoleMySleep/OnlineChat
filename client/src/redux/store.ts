import { configureStore } from '@reduxjs/toolkit';
import author from './componentReducers/login';
import messages from './componentReducers/messages';

export default configureStore({
  reducer: {
    messages,
    author
  }
}); 