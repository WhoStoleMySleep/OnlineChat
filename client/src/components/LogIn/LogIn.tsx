import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import useClassOpen from '../../hooks/useClassOpen/useClassOpen';
import useInputChange from '../../hooks/useInputChange/useInputChange';
import useSubmitAuthor from '../../hooks/useSubmitAuthor/useSubmitAuthor';
import { setAuthor } from '../../redux/componentReducers/login';
import './LogIn.scss';

function LogIn() {
  const getLocalStorageAuthor = localStorage.getItem('author');
  const dispatch = useDispatch();
  const [authorState, setAuthorState] = useState(
    getLocalStorageAuthor || ''
  );

  useEffect(() => {
    dispatch(setAuthor(authorState));
  }, []);

  const { onClick: onFromOpen } = useClassOpen('log-in__form', 'open');
  const { onSubmit: onFormSubmit } = useSubmitAuthor('onSubmit', 'log-in__form', 'open', authorState, setAuthor);
  const { onClick: onSubmitButtonClick } = useSubmitAuthor('onClick', 'log-in__form', 'open', authorState, setAuthor);
  const { onChange: onInputChange } = useInputChange(setAuthorState);

  return (
    <div className="log-in">
      <button
        type="button"
        className="log-in__open-form"
        onClick={onFromOpen}
        data-testid="login-open-button"
      >
        Log-In
      </button>
      <form
        action=""
        className="log-in__form"
        onSubmit={onFormSubmit}
        data-testid="login-form"
      >
        <input
          type="text"
          className="log-in__author"
          placeholder="Enter your nickname"
          maxLength={25}
          value={authorState}
          onChange={onInputChange}
          data-testid="login-name-input"
        />
        <br />
        <button
          type="button"
          className="log-in__close-form"
          onClick={onSubmitButtonClick}
          data-testid="login-submit-button"
        >
          Ð¡onfirm
        </button>
      </form>
    </div>
  );
}

export default LogIn;
