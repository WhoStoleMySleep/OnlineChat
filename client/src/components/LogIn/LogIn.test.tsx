import '@testing-library/jest-dom';
import {
  render, screen
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import LogIn from './LogIn';

describe('LogIn Test', () => {
  it('Login snapshot', () => {
    render(
      <Provider store={store}>
        <LogIn />
      </Provider>
    );

    const logIn = screen.getByTestId('logIn');

    expect(logIn).toMatchSnapshot();
  });

  it('LogIn name input', async () => {
    render(
      <Provider store={store}>
        <LogIn />
      </Provider>
    );

    const input = screen.getByTestId('input');

    expect(input).toHaveValue('');

    userEvent.clear(input);

    expect(input).toHaveValue('');

    await userEvent.type(input, '123');

    expect(input).toHaveValue('123');

    userEvent.clear(input);
    await userEvent.type(input, '99999999999999999999999999'); // 26 symbol

    expect(input).toHaveValue('9999999999999999999999999'); // 25 symbol
  });
});
