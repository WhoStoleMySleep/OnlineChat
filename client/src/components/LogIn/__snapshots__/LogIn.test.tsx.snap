// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn Test Login snapshot 1`] = `
<div
  class="log-in"
  data-testid="logIn"
>
  <button
    class="log-in__open-form"
    data-testid="openButton"
    type="button"
  >
    Log-In
  </button>
  <form
    action=""
    class="log-in__form"
    data-testid="form"
  >
    <input
      class="log-in__author"
      data-testid="input"
      maxlength="25"
      placeholder="Enter your nickname"
      type="text"
      value=""
    />
    <br />
    <button
      class="log-in__close-form"
      data-testid="submitButton"
      type="submit"
    >
      Ð¡onfirm
    </button>
  </form>
</div>
`;
